src/metadata.rs

```diff
@@ -0,0 +1,109 @@
+use grpc_sys::{self, GrpcMetadataArray};
+use std::ffi::CString;
+use libc::c_char;
+#[allow(unused)]
+use std::ascii::AsciiExt;
+use std::slice;
+
+/// Builder used to construct a MetadataArray value.
+pub struct MetadataArrayBuilder {
+    entries: Vec<(*mut c_char, *mut c_char, usize)>
```

Comment:
Memory leak if `build` is not called at the end. Why not keep it as `CString`?

Reply:
Good point. Maybe it's better to represent it as Vec<(CString, Vec<u8>). The key is expected to be a null terminated c-style string so CString seems appropriate for that, while the value is treated as an opaque byte array so better to keep it as Vec<u8>. 

[relevant commit](https://github.com/pingcap/grpc-rs/pull/135/commits/345821e1365ace32d0df70242d3a63bc21ae7eae)